src/ImageSharp/Common/Helpers/SimdUtils.Avx2Intrinsics.cs

```diff
@@ -91,6 +91,190 @@ public static class Avx2Intrinsics
+                }
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            internal static void PackBytesToUInt24Reduce(
+                    ref ReadOnlySpan<byte> channel0,
+                    ref ReadOnlySpan<byte> channel1,
+                    ref ReadOnlySpan<byte> channel2,
+                    ref Span<byte> dest)
+            {
+                DebugGuard.IsTrue(channel0.Length == dest.Length, nameof(channel0), "Input spans must be of same length!");
+                DebugGuard.IsTrue(channel1.Length == dest.Length, nameof(channel1), "Input spans must be of same length!");
+                DebugGuard.IsTrue(channel2.Length == dest.Length, nameof(channel2), "Input spans must be of same length!");
+
+                if (Avx2.IsSupported)
+                {
+                    int remainder = ImageMaths.ModuloP2(channel0.Length, Vector256<byte>.Count);
+                    int adjustedCount = channel0.Length - remainder;
+
+                    if (adjustedCount > 0)
+                    {
+                        channel0 = channel0.Slice(adjustedCount);
+                        channel1 = channel1.Slice(adjustedCount);
+                        channel2 = channel2.Slice(adjustedCount);
+                        dest = dest.Slice(adjustedCount);
+
+                        PackBytesToUInt24(
+                            channel0,
+                            channel1,
+                            channel2,
+                            dest);
+                    }
+                }
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            internal static void PackBytesToUInt24(
+                    ReadOnlySpan<byte> channel0,
+                    ReadOnlySpan<byte> channel1,
+                    ReadOnlySpan<byte> channel2,
+                    Span<byte> dest)
+            {
+                VerifySpanInput(channel0, dest, Vector256<byte>.Count);
+                VerifySpanInput(channel1, dest, Vector256<byte>.Count);
+                VerifySpanInput(channel2, dest, Vector256<byte>.Count);
+
+                int n = dest.Length / Vector256<byte>.Count;
+
+                ref Vector256<byte> source0Base =
+                    ref Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(channel0));
+                ref Vector256<byte> source1Base =
+                    ref Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(channel1));
+                ref Vector256<byte> source2Base =
+                    ref Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(channel2));
+
+                ref Vector256<byte> destBase =
+                    ref Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(dest));
+
+                Vector256<byte> s0Mask0 = Vector256.Create(0, -1, -1, 1, -1, -1, 2, -1, -1, 3, -1, -1, 4, -1, -1, 5, -1, -1, 6, -1, -1, 7, -1, -1, 8, -1, -1, 9, -1, -1, 10, -1).AsByte();
```

Comment:
They'll direct you to use `Vector256.Create` like this over in the runtime repo because they've fixed the JIT to recognize it in 5.0.  It's exceedingly expensive in 3.x, though, and since 3.1 has 2.5 years of LTS left, I would personally keep using the `ReadOnlySpan<byte>` trick to load these from static data.

@JimBobSquarePants do you have a preference there?

Reply:
Yeah, let's use the ROS trick here. We want to try to max out on all supported frameworks.